alacritty/src/wayland_theme.rs

```diff
@@ -1,26 +1,30 @@
-use glutin::platform::unix::{ButtonState, Theme as WaylandTheme};
+use glutin::platform::unix::{ARGBColor, Button, ButtonState, Element, Theme as WaylandTheme};
 
 use alacritty_terminal::config::Colors;
-use alacritty_terminal::term::color::{Rgb, DIM_FACTOR};
+use alacritty_terminal::term::color::Rgb;
 
 #[derive(Debug, Clone)]
 pub struct AlacrittyWaylandTheme {
-    pub background: Rgb,
-    pub foreground: Rgb,
-    pub dim_foreground: Rgb,
-    pub hovered_close_icon: Rgb,
-    pub hovered_maximize_icon: Rgb,
-    pub hovered_minimize_icon: Rgb,
+    pub background: ARGBColor,
+    pub foreground: ARGBColor,
+    pub dim_foreground: ARGBColor,
+    pub hovered_close_icon: ARGBColor,
+    pub hovered_maximize_icon: ARGBColor,
+    pub hovered_minimize_icon: ARGBColor,
 }
 
 impl AlacrittyWaylandTheme {
     pub fn new(colors: &Colors) -> Self {
-        let hovered_close_icon = colors.normal().red;
-        let hovered_maximize_icon = colors.normal().green;
-        let hovered_minimize_icon = colors.normal().yellow;
-        let foreground = colors.primary.foreground;
-        let background = colors.primary.background;
-        let dim_foreground = colors.primary.dim_foreground.unwrap_or(foreground * DIM_FACTOR);
+        let hovered_close_icon = colors.normal().red.into_rgba();
+        let hovered_maximize_icon = colors.normal().green.into_rgba();
+        let hovered_minimize_icon = colors.normal().yellow.into_rgba();
+        let foreground = colors.search_bar_foreground().into_rgba();
+        let background = colors.search_bar_background().into_rgba();
+
+        let mut dim_foreground = foreground;
+
+        // Blend with background with 0.5 for opacity.
+        dim_foreground.a = 127;
```

Comment:
Could we use DIM_FACTOR here? Or does that not work out well? In theory there's no direct relation, but it might be nice.

Otherwise we should define a new constant.

Reply:
it doesn't work nice with light themes well, those weren't opaque enough...

Reply:
Huh, mind sharing a screenshot just out of curiosity? You're saying that it's not clear enough that the window is inactive?

Reply:
That's 0.5

![05](https://user-images.githubusercontent.com/27620401/95143785-b5f67100-077f-11eb-8ef1-ae80852c9cf6.png)


That's DIM_FACTOR:

![dim](https://user-images.githubusercontent.com/27620401/95143805-bf7fd900-077f-11eb-9b09-b0b72b425704.png)

And that's without dimming at all.

![nodim](https://user-images.githubusercontent.com/27620401/95143861-e63e0f80-077f-11eb-9ebc-d2600aa3e696.png)


Like it's really not that noticeable when you don't dim at all or you dim with `DIM_FACTOR`. The `0.5` works, since it's the same amount of change for dark/light backgrounds, hence 0.5.

Reply:
>Otherwise we should define a new constant.

I don't see a value in defining new constant, since it's used in one place for one specific reason and has a comment above it already why it's done that way. If we've used multiple places for that, sure, but it's only one in constructor, and the constant will be local to a file anyway...

Reply:
I'd still like it to be a constant at the top of the file, that makes it much more valueable than some random value somewhere. There shouldn't be a need to add a comment.

Also do you want to report the transparency bug with the icons or should I?

Reply:
I don't see any bug, all three images are different (maybe not that noticeable, but when you look at decorations it's noticeable).

As you can see in https://github.com/Smithay/client-toolkit/blob/0bbda781acc40798b31cd2b08a454b432ecf6e85/src/window/concept_frame.rs#L994 it seems fine, it's just 0.66 and 0.5 difference it actually not that small if you think about it.

Reply:
Zoom in on the arrows, transparency isn't applied correctly. They seem to be multiple different shapes that overlap.

Reply:
![tmp](https://user-images.githubusercontent.com/8886672/95207123-0b11a180-07d7-11eb-8292-d5baa2983101.png)

Reply:
Ah, that one, well, I guess we can say that it blended something that it shouldn't blend. You can report that to sctk, I guess.

Reply:
Not sure if it's sctk or andrew, but I'll report to sctk and levanx can then move it I suppose.

Reply:
I've created a bug (https://github.com/Smithay/client-toolkit/issues/161), I'd like a constant, but then everything should be good to go.